packages/mirinae/src/data-display/avatar/PAvatar.vue

```diff
@@ -0,0 +1,110 @@
+<script setup lang="ts">
+import { computed, reactive } from 'vue';
+
+import { get, isNull } from 'lodash';
+
+import type { SizeType, ColorType } from '@/data-display/avatar/type';
+import { AVATAR_SIZE, AVATAR_COLOR, ICON_SIZE } from '@/data-display/avatar/type';
+import PLazyImg from '@/feedbacks/loading/lazy-img/PLazyImg.vue';
+import PI from '@/foundation/icons/PI.vue';
+
+
+import colors from '@/styles/colors.cjs';
+
+interface AvatarProps {
+    color?: ColorType;
+    size?: SizeType;
+    icon?: string;
+    imgSrc?: string;
+}
+
+const props = withDefaults(defineProps<AvatarProps>(), {
+    imgSrc: '',
+    icon: 'ic_avatar-filled',
+    color: AVATAR_COLOR.INDIGO_200,
+    size: AVATAR_SIZE.MD,
+});
+
+const state = reactive({
+    imgSrc: computed(() => props.imgSrc),
+    iconName: computed(() => props.icon),
+    iconSize: computed(() => ICON_SIZE[props.size]),
+    avatarInlineStyles: computed(() => {
+        const iconSize = ICON_SIZE[props.size];
+
+        return {
+            width: iconSize,
+            height: iconSize,
+        };
+    }),
+    wrapperInlineStyles: computed(() => {
+        const defaultStyle = { backgroundColor: AVATAR_COLOR.INDIGO_200 };
+
+        const colorNum = props.color.match(/\d{3}/)?.[0];
+
+        if (isNull(colorNum)) {
+            return defaultStyle;
+        }
```

Comment:
todo: `isNull` only checks `null`. Why don't you use `isNil` to check `undefined` as well?

Reply:
or we can just use 
```typescript
if (colorNum)
```

Reply:
This is a mistake, I thought colorNum only returned null